src/screens/CreateEventPage.js

```diff
@@ -27,20 +41,20 @@ class CreateEventPage extends Component {
 
       //Name
       if (!fields["eventNameText"]) {
-         this.state.formIsValid.setState(false);
+         this.state.formIsValid=false;
```

Comment:
This is probably what it was yelling at you for modifying state directly. It should be `this.setState({formIsValid: false})`